pychromecast/socket_client.py

```diff
@@ -489,6 +489,11 @@ def new_cast_status(self, status: CastStatus) -> None:
         self.destination_id = status.transport_id
         self.session_id = status.session_id
 
+        # When a new cast status is received, give Audible time to load
+        if status.app_id == "25456794":
+            self.logger.debug("Detected Audible connection. Sleeping for 1s")
+            time.sleep(1)
```

Comment:
This is not really OK way to fix a race. Is there another message we can wait for instead?

Reply:
i'm not sure. any ideas you have, or example code you can point me to? i'll try whatever i can.

Reply:
Issue https://github.com/home-assistant-libs/pychromecast/issues/738 talks about the Sirius XM app misbehaving, but your comment in the PR talks about Audible. Could you clarify the relationship between the two apps? Or are they the same?

Also, after going through the comments in https://github.com/home-assistant-libs/pychromecast/issues/738, it's not fully clear to me what pychromecast does which offends the app so much that it terminates, only that adding a delay fixes the problem.

Have you been able to determine *how* the delay fixes it?

If you run pychromecast with debug enabled, it will dump out all messages sent to and received from the pychromecast, have you done that?

Also, I'm not sure how you've been debugging. But if you want a simple isolated test case, maybe this example script is enough to trigger (and debug) the problem:
```
python3 -m examples.simple_listener_example --cast 'Living room' --show-debug
```

Reply:
Ah, there's already a comment in #738 about using the simple listener example to trigger the problem: https://github.com/home-assistant-libs/pychromecast/issues/738#issuecomment-1650581622

If you could do that, both with and without your workaround, that would be really useful.

Reply:
```sh
git switch master
python3 -m examples.simple_listener_example --cast 'Garage speakers' --show-debug > workspace/master.std.txt 2> workspace/master.err.txt
```
[master.err.txt](https://github.com/user-attachments/files/16825091/master.err.txt)
[master.std.txt](https://github.com/user-attachments/files/16825092/master.std.txt)


```sh
git switch geoffoxholm:users/geoff/audible_debug
python3 -m examples.simple_listener_example --cast 'Garage speakers' --show-debug > workspace/patch.std.txt 2> workspace/patch.err.txt
```
[patch.err.txt](https://github.com/user-attachments/files/16825094/patch.err.txt)
[patch.std.txt](https://github.com/user-attachments/files/16825095/patch.std.txt)

Reply:
There's a jump in `requestId` in messages sent by pychromecast to the device which I can't explain, which is really odd because we only increment the `requestId` when calculating it here: https://github.com/home-assistant-libs/pychromecast/blob/449c0b52261804a2a1077700d6653d69cf902691/pychromecast/socket_client.py#L855-L856

Have you removed any lines from the logs?

Reply:
To be sure, could you change here to log also the pings, both on master and your patch branch, and collect new logs: https://github.com/home-assistant-libs/pychromecast/blob/449c0b52261804a2a1077700d6653d69cf902691/pychromecast/socket_client.py#L873-L882

https://github.com/home-assistant-libs/pychromecast/blob/449c0b52261804a2a1077700d6653d69cf902691/pychromecast/socket_client.py#L689-L697

Reply:
```sh
git switch master
git pull
# make edits
git diff > workspace/master.diff.txt
python3 -m examples.simple_listener_example --cast 'Garage speakers' --show-debug > workspace/master.std.txt 2> workspace/master.err.txt
```
[master.diff.txt](https://github.com/user-attachments/files/16875720/master.diff.txt)
[master.err.txt](https://github.com/user-attachments/files/16875724/master.err.txt)
[master.std.txt](https://github.com/user-attachments/files/16875726/master.std.txt)

```sh
git stash # stash edits
git switch users/geoff/audible_debug 
git stash pop # pop edits
git diff master > workspace/branch.diff.txt
python3 -m examples.simple_listener_example --cast 'Garage speakers' --show-debug > workspace/branch.std.txt 2> workspace/branch.err.txt
```
[branch.diff.txt](https://github.com/user-attachments/files/16875735/branch.diff.txt)
[branch.err.txt](https://github.com/user-attachments/files/16875737/branch.err.txt)
[branch.std.txt](https://github.com/user-attachments/files/16875739/branch.std.txt)

Reply:
> Issue https://github.com/home-assistant-libs/pychromecast/issues/738 talks about the Sirius XM app misbehaving, but your comment in the PR talks about Audible. Could you clarify the relationship between the two apps? Or are they the same?

They are different apps. In the end the issue with Sirius XM was "fixed" (avoided) when the app was redesigned. It seems the Audible app has a similar pathology to the one that was present in the Sirius XM app, but no longer is.

Reply:
Hi @emontnemery,

Any insight from the last logs you requested from @geoffoxholm?

I also have similar problem and setup and would like to contribute so that Audible works when pychromecast is active on the same network. 

I can test the changes fix some logic.

As of now the sleep tricks works, but I understand in your judgement it wont ever be merged  since it's an ugly trick to fix race condition.

What is the better way to handle this? To me this behaves like an app quirk,and should be handled as so.